### DISCUSSION TITLE
boundary faces

### DISCUSSION THREAD

**Discussion by edljk** [Ferrite-FEM/Ferrite.jl#565] [Ideas]
I wanted to share an observation that you may have in mind but which could get some troubles for a naive user.
Defining faces using a characteristic function of the boundary like

```addfaceset!(grid, "∂Ω", x -> onboundary(x))```

may select faces that are actually not on the boundary.  See the left bottom corner on the illustration 

![boundary](https://user-images.githubusercontent.com/14992507/208245738-d8832254-3ee0-472a-b9c4-a0928b66a8fb.png)

Perhaps it could be interesting to define an `addfaceset!` function which only selects boundary faces which is perhaps the most common expected behavior defining a faceset.

**Comment by edljk** [comment]
Just for the record: to avoid to select interior faces, I always intersect my `facesets` with the boundary `faceset` (which is identified as the set of faces which are associated to only one cell)

**Comment by termi-official** [comment]
We should be able to resolve this with the help of the topology. A boundary face is by definition a local face with no corresponding neighbor.

**Comment by termi-official** [comment]
How have you generated and loaded the grid?

**Comment by edljk** [comment]
I am using the `Triangle` mesh library to get nodes and cells.
The grid is loaded by a direct call 
```julia  
grid = Ferrite.Grid(cells, nodes)
 ```

**Comment by graphitical** [comment]
I'm interested in this idea and the [open issue](https://github.com/Ferrite-FEM/Ferrite.jl/issues/595). 

How can I check/plot the facesets individually? None of the FerriteViz plots seem to take a `facesets` kwarg. Maybe this is the first step, implementing a simple way to show facesets on a wireframe?

**Reply by termi-official** [reply]
Hi Daniel, thanks for reaching out. I had no time to implement any kind of infrastructure visualization for boundary conditions yet but I opened an issue [here](https://github.com/Ferrite-FEM/FerriteViz.jl/issues/14) for the facesets. Should be straight forward to implement if you are interested in these. If you run into issues feel free to reach me again.
However, while it is nice to have visualizations at hand, #595 should be straightforward to solve without any kind of visualization. So, if you are not interested in visualization and only in FEM, then feel free to to skip https://github.com/Ferrite-FEM/FerriteViz.jl/issues/14 . :)

**Comment by termi-official** [comment]
This should be fixed in https://github.com/Ferrite-FEM/Ferrite.jl/pull/606 .
